src/prefect/_internal/concurrency/event_loop.py

```diff
@@ -46,4 +56,23 @@ def wrapper() -> None:
                 raise
 
     __loop.call_soon_threadsafe(wrapper)
-    return future.result()
+    return future
```

Comment:
Can't say I understand what's going on here, but I'm ok with that.

Reply:
These utilities are kind of dumb, but basically `AbstractEventLoop.call_soon_threadsafe` returns a `Handle` which is the most useless object around town — you can cancel it and that's it. In cases where we might like, actually want to know what our function returned or wait for the result of our function, we need to return something else. To accomplish this, we wrap the function that we submit to `call_soon_threadsafe` and use a threading `Future` to capture the return value.

Reply:
I've added some additional documentation for these functions in the next PR